### ISSUE TITLE
bug: ASF SQS provider is very slow

### ISSUE THREAD

**Issue by jansepke** [localstack/localstack#6093]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

we are using the latest localstack docker image for our integration tests. Since this commit ad2763d89e9ce934ee0655112ca16f91889dc5cf our integration tests are failing because this new SQS provider seems to be very slow.

### Expected Behavior

SQS working as before commit ad2763d89e9ce934ee0655112ca16f91889dc5cf

### How are you starting LocalStack?

With a `docker run` command

### Steps To Reproduce

TBD

### Environment

```markdown
- OS: Ubuntu 22.04
- LocalStack: docker localstack/localstack:latest
```

**Comment by thrau** [comment]
I ran the benchmark script from #3998 again on the latest version, and get these results (which look very good)

```
 % python benchmark.py
Using SQS endpoint: http://localhost:4566
number of events to send: 1000
sending in parallel with 50 concurrent connections/workers
sending finished, avg duration: 0.212s, total duration: 4.348
receiving+deleting sequentially - one by one
received: 1000 of 1000
receiving+deleting finished, avg duration receive: 0.007s, avg duration delete: 0.007s, total duration: 13.518
```

Here are the results when starting localstack with `PROVIDER_OVERRIDE_SQS=legacy` (which starts the old provider)
```
 % python benchmark.py
Using SQS endpoint: http://localhost:4566
number of events to send: 1000
sending in parallel with 50 concurrent connections/workers
sending finished, avg duration: 0.712s, total duration: 14.536
receiving+deleting sequentially - one by one
received: 1000 of 1000
receiving+deleting finished, avg duration receive: 0.027s, avg duration delete: 0.014s, total duration: 40.721
```

**Comment by shadowhand** [comment]
FWIW, I have the opposite result: the new SQS provider is significantly faster than the legacy one. The legacy one would often timeout while waiting to receive messages, even with a 5s wait. The new SQS provider does have some issues, but in my experience performance is **not** one.

**Comment by wimax-grapl** [comment]
@jansepke, are you specifying WaitTimeSeconds? If so this may be https://github.com/localstack/localstack/issues/5824

**Comment by thrau** [comment]
@jansepke are you using HTTPS? in that case it may be related to #5499 which we are working on. if you could provide additional information about your configuration and application i'm happy to try to reproduce it.

**Comment by jansepke** [comment]
thank you we are using WaitTimeSeconds everywhere. I just tried out the latest docker images and the problems are gone :+1:

**Comment by jansepke** [comment]
sorry

**Comment by jansepke** [comment]
yes we are using WaitTimeSeconds mostly, but the problem is still there when using the latest localstack version from docker. We will try to narrow it down and provide a reproduceable issue.
